contrib/gogearbox/gearbox.v1/internal/gearboxutil/fashttpcarrier.go

```diff
@@ -0,0 +1,61 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package gearboxutil // import "gopkg.in/DataDog/dd-trace-go.v1/contrib/gogearbox/gearbox/internal/gearboxutil"
+
+// MTOFF: This will be useful if we add a fasthttp integration. Might also be useful for the fiber integration.
+// But should it be separated under a fashttputils....
+
+import (
+	"strings"
+
+	"github.com/valyala/fasthttp"
+)
+
+// FasthttpCarrier implements tracer.TextMapWriter and tracer.TextMapReader on top
+// of fasthttp's RequestHeader object, allowing it to be used as a span context carrier for
+// distributed tracing.
+type FasthttpCarrier struct {
+	ReqHeader *fasthttp.RequestHeader
+}
+
+// ForeachKey iterates over fasthttp request header keys and values
+func (f *FasthttpCarrier) ForeachKey(handler func(key, val string) error) error {
+	keys := f.ReqHeader.PeekKeys()
+	for _, key := range keys {
+		sKey := string(key)
+		vs := f.ReqHeader.PeekAll(sKey)
+		for _, v := range vs {
+			if err := handler(sKey, string(v)); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// func (f *FasthttpCarrier) ForeachKey(handler func(key, val string) error) (err error) {
+// 	f.ReqHeader.VisitAll(func(k, v []byte) {
+// 		if e := handler(string(k), string(v)); e != nil {
+// 			err = e
+// 			// Break here to quick return
+// 		}
+// 	})
+// 	return err
+// }
```

Comment:
**For the Go team:** 
In the case that there is a need to optimize in the future, it may be handy to make this idea readily available to the person that works on the optimization. Is there anywhere that I could "store" this information in the case that a future optimization is needed? Maybe create a card on the APM Go Board with this info and all the other optimization points I left in the comments?